csi.proto

```diff
@@ -1634,3 +1677,165 @@ message NodeExpandVolumeResponse {
   // The capacity of the volume in bytes. This field is OPTIONAL.
   int64 capacity_bytes = 1;
 }
+message GroupControllerGetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+message GroupControllerGetCapabilitiesResponse {
+  // All the capabilities that the group controller service supports.
+  // This field is OPTIONAL.
+  repeated GroupControllerServiceCapability capabilities = 1;
+}
+
+// Specifies a capability of the group controller service.
+message GroupControllerServiceCapability {
+  message RPC {
+    enum Type {
+      UNKNOWN = 0;
+
+      // Indicates that the group controller plugin supports
+      // creating, deleting, and getting details of a volume
+      // group snapshot.
+      CREATE_DELETE_GET_VOLUME_GROUP_SNAPSHOT = 1
+      [(alpha_enum_value) = true];
+    }
+
+    Type type = 1;
+  }
+
+  oneof type {
+    // RPC that the controller supports.
+    RPC rpc = 1;
+  }
+}
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The suggested name for the group snapshot. This field is REQUIRED
+  // for idempotency.
+  // Any Unicode string that conforms to the length limit is allowed
+  // except those containing the following banned characters:
+  // U+0000-U+0008, U+000B, U+000C, U+000E-U+001F, U+007F-U+009F.
+  // (These are control characters other than commonly used whitespace.)
+  string name = 1;
+
+  // volume ids of the source volumes to be snapshotted together.
+  // This field is REQUIRED.
+  repeated string source_volume_ids = 2;
+
+  // Secrets required by plugin to complete
+  // ControllerCreateVolumeGroupSnapshot request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  // The secrets provided in this field SHOULD be the same as
+  // the secrets provided in ControllerDeleteVolumeGroupSnapshot
+  // and ControllerGetVolumeGroupSnapshot requests for the same
+  // group snapshot unless if secrets are rotated after the
+  // group snapshot is created.
+  map<string, string> secrets = 3 [(csi_secret) = true];
+
+  // Volume secrets required by plugin to complete volume group snapshot
+  // creation request.
+  // This field is OPTIONAL.
+  repeated VolumeSecret volume_secrets = 4;
```

Comment:
@xing-yang both of these secret ( volume level and general) field explanation looks redundant ( as both mention create group snapshotrequest) to me. do we need to make it bit more clear why we have 2 level of secret fields ?

Reply:
@humblec This volume_secrets is added specifically to support volume-level secrets. We have one field "secrets" to create a group snapshot, but what if the secrets required to access each volume is different?  This is something that needs your confirmation as Ceph supports volume-level secrets.  Do you mean one "Secrets" for creating group snapshot is enough and there is no need to have a volume_secrets?

Reply:
@xing-yang so far the creds are used from high level RPC service level configuration. Considering required capabilities are attached to the same creds,  those were good enough and can be used to connect to backend cluster and can operate even on low level resources like volumes, hence good enough to satisfy the requirement came across so far. However, imo, having a provision even in volume level is good so that consumers have the choice of using one or both.   my comment was mainly about just making the comment section of these fields bit more clear on the aspect that, it is an additional configuration option for consumers if they want to have volume level secret supply for operating on the volume.

Reply:
Added comments to clarify why volume_secrets are needed: https://github.com/container-storage-interface/spec/pull/519/commits/3bd6f95368c6d6c81409ab5915dc5c5619d490b8